### ISSUE TITLE
ValidationSchema does not respect configuration provided to `load`

### ISSUE THREAD

**Issue by ssilve1989** [nestjs/config#311]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Given the following code: 
```typescript
import { Module } from '@nestjs/common';
import { ConfigModule } from '@nestjs/config';
import Joi from '@hapi/joi';

const configuration = () => ({
  foo: 'bar',
});

@Module({
  imports: [
    ConfigModule.forRoot({
      validationSchema: Joi.object({
        foo: Joi.string().required(),
      }),
      load: [configuration],
    }),
  ],
})
export class AppModule {}
```
The following error is produced 
```shell
Error: Config validation error: "foo" is required
    at Function.forRoot (/dev/playground/node_modules/@nestjs/config/dist/config.module.js:43:23)
    at /dev/playground/src/app.module.ts:11:18
    at Object.<anonymous> (/dev/playground/src/app.module.ts:44:3)
    at Module._compile (internal/modules/cjs/loader.js:1200:30)
    at Module.m._compile (/Users/stevensilvestri/.config/yarn/global/node_modules/ts-node/src/index.ts:473:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/stevensilvestri/.config/yarn/global/node_modules/ts-node/src/index.ts:476:12)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Module.require (internal/modules/cjs/loader.js:1089:19)
```

## Expected behavior

No error should be thrown

## Minimal reproduction of the problem with instructions

see snippet above

## What is the motivation / use case for changing the behavior?

The configuration is providing the value, it should not fail validation when it is provided.

## Environment

<pre><code>
Nest version: 7.0.13

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
`ValidationSchema` is not intended to validate partial configurations https://github.com/nestjs/config/issues/209#issuecomment-625765057

You can put the validation logic directly in the factory.

**Comment by vtgn** [comment]
Hi!

I reopen this issue because I got exactly the same problem.
Indeed, my "validate" function failed to validate the object built by the function given in the "load" field. By adding several console.log, I noticed the function given in the "load" field was called after the starting of the nest application, and so after the calling of the validate function.
I don't understand either why the files given in the "envFilePath" field are loaded before the calling of the "validate" method, and not the objects given in the "load" field.
This is totally illogical! And besides, the documentation says absolutely nothing on this subject.

When @kamilmysliwiec answered we can put the validation logic directly in the factory, we can also do the same with the files given in the "envFilePath", or with the environment variables of the system! This answer is just a nonsense, exactly the same way the "load" feature is implemented.

An API is made to facilitate the developers work, not to implement strange features that are completely illogical in certain cases with the goal of this API.

Regards.
